### PR TITLE
f(wall_following) initial commit

### PR DESCRIPTION
**Pull request by SammyOina**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (rodneyosodo) [inline .vscode/settings.json]: Remove this

File: platformio.ini [hunk 2]
```diff
@@ -0,0 +1,14 @@
+; PlatformIO Project Configuration File
```
Comment (rodneyosodo) [inline platformio.ini]: Is this a library of a project?

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Structure it a a library not a project. Here is the reference https://github.com/0804robot/driver/pull/3

**Review by rodneyosodo — APPROVED** [review]

**Review by kimanimichael — COMMENTED** [review]
If we are solving a maze, we may have a scenario where there's a wall in all directions and need to U-turn. Can we take care of this?

**Comment by SammyOina** [comment]
> If we are solving a maze, we may have a scenario where there's a wall in all directions and need to U-turn. Can we take care of this?

The current logic only relies on two ultrasonic sensors but we can include that in the next iteration

**Review by WashingtonKK — APPROVED** [review]

**Pull request merged**